server/pulp/server/managers/consumer/applicability.py

```diff
@@ -540,7 +532,16 @@ def remove_orphans():
         #
 
         # Find the profile hashes that exist in current UnitProfiles
-        active_profile_hashes = UnitProfile.get_collection().distinct('profile_hash')
+        unit_profile_coll = UnitProfile.get_collection()
+        active_profile_hashes = unit_profile_coll.distinct('profile_hash')
+
+        # Find if there are any empty consumer profiles, remove them from the active ones because
+        # applicability for those is not needed. Max is one profile per content type.
```

Comment:
2 content types are supported, 'rpm' and 'modulemd', so max there will be 2 profiles to remove. In the majority of cases it will be 1 because 'rpm' profiles are almost never empty.